dcpy/models/dataset.py

```diff
@@ -0,0 +1,32 @@
+from dcpy.models.base import SortedSerializedBase
+from typing import Literal
+
+COLUMN_TYPES = Literal[
+    "text",
+    "integer",
+    "decimal",
+    "number",  # TODO: Need to delete. Keeping it now for compatibility with metadata files
+    "geometry",
+    "bool",
+    "bbl",
+    "date",
+    "datetime",
+]
+
+
+# TODO: extend/modify Checks model
+class Checks(SortedSerializedBase):
+    is_primary_key: bool | None = None
+    non_nullable: bool | None = None
+
+
+class Column(SortedSerializedBase, extra="forbid"):
+    """
+    An extensible base class for defining column metadata in ingest and product templates.
+    """
+
+    id: str
+    data_type: COLUMN_TYPES | None = None
+    description: str | None = None
+    is_required: bool = True
+    checks: Checks | None = None
```

Comment:
Gut impression here - seems that `checks` should maybe be a list? Where then checks might take form
```
- name (or id): {name}
  parameter: value
```
Or something like that. Could just be a list of strings too. Will think more (and look at our old discussions) and revisit tomorrow

Reply:
Not sure yet, I'm planning to focus on Checks in next PR. In this PR, I just moved the existing Checks obj Alex implemented to this file. If I don't do that, I run into an error where imports between `dataset.py` & `metadata_v2.py` are circular.

Reply:
Yeah I think that makes sense to keep for later, since we're not actually attempting to implement checks yet.